### PR TITLE
User experience refactor

### PR DESCRIPTION
**Pull request by dmitrymarchuk**
This PR continues to remove worldUsers from the codebase. Highlights from changes:

- Audience worldUsers code was just commented out
- code related to Twilio `Participants` now fetches user data as `Participant`s connect (relates to all venues with video, except Auditorium)
- JazzBar, ConversationSpace, FriendshipPage now use `seatedTableUsers` subcollection to show users seated at tables
- various refactorings

### CODE REVIEW

File: src/api/profile.ts [hunk 1]
```diff
@@ -2,14 +2,22 @@ import Bugsnag from "@bugsnag/js";
...
+import { WithId, WithVenueId } from "utils/id";
 
 export const getUserRef = (userId: string) =>
   firebase.firestore().collection("users").doc(userId);
 
+export const getUser = async (userId: string): Promise<WithId<User>> => {
+  const snapshot = await getUserRef(userId).get();
+  return {
+    ...snapshot.data(),
+    id: snapshot.id,
```
Comment (mike-lvov) [inline src/api/profile.ts:17]: withId

File: src/components/organisms/Room/Room.tsx [hunk 2]
```diff
@@ -102,61 +90,32 @@ const Room: React.FC<RoomProps> = ({
...
-    const firestore = firebase.firestore();
-    await firestore
-      .doc(doc)
-      .update(update)
-      .catch(() => {
-        firestore.doc(doc).set(update);
-      });
-    setSeatedAtTable && setSeatedAtTable("");
-  }, [firebase, profile, setSeatedAtTable, user, venueName]);
+    if (!userId) return;
```
Comment (mike-lvov) [inline src/components/organisms/Room/Room.tsx:93]:
```suggestion
    if (!userId) return;
    
```

File: src/components/organisms/Room/Room.tsx [hunk 3]
```diff
@@ -102,61 +90,32 @@ const Room: React.FC<RoomProps> = ({
...
+    await unsetTableSeat(userId, { venueId });
+    setSeatedAtTable?.("");
```
Comment (mike-lvov) [inline src/components/organisms/Room/Room.tsx:95]: why do we need this?

**Review by mike-lvov — APPROVED** [review]
Looks good!

File: src/components/organisms/Room/Room.tsx (hunk 2 above)
Comment (dmitrymarchuk) [reply src/components/organisms/Room/Room.tsx]: fixed in https://github.com/sparkletown/sparkle/pull/2338/commits/684a3f91796747463ab8f37a3497625fd9d2b89e

File: src/api/profile.ts (hunk 1 above)
Comment (dmitrymarchuk) [reply src/api/profile.ts]: fixed in https://github.com/sparkletown/sparkle/pull/2338/commits/684a3f91796747463ab8f37a3497625fd9d2b89e

**Pull request merged**